File: apollo-compiler/src/test/graphql/com/example/custom_scalar_type/type/CustomType.java
package com.example.custom_scalar_type.type;

import com.apollographql.apollo.api.ScalarType;
import java.lang.Class;
import java.lang.Integer;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Date;
import javax.annotation.Generated;

@Generated("Apollo GraphQL")
public enum CustomType implements ScalarType {
  DATE {
    @Override
    public String typeName() {
      return "Date";
    }

    @Override
    public Class javaType() {
      return Date.class;
    }
  },

  UNSUPPORTEDTYPE {
    @Override
    public String typeName() {
      return "UnsupportedType";
    }

    @Override
    public Class javaType() {
      return Object.class;
    }
  },

  URL {
    @Override
    public String typeName() {
      return "URL";
    }

    @Override
    public Class javaType() {
      return String.class;
    }
  },

  ID {
    @Override
    public String typeName() {
      return "ID";
    }

    @Override
    public Class javaType() {
      return Integer.class;
    }
  }
}

File: apollo-compiler/src/test/graphql/com/example/custom_scalar_type/type/CustomType.kt
package com.example.custom_scalar_type.type

import com.apollographql.apollo.api.ScalarType
import java.lang.Class
import javax.annotation.Generated
import kotlin.String

@Generated("Apollo GraphQL")
enum class CustomType : ScalarType {
    DATE {
        override fun typeName(): String = "Date"

        override fun javaType(): Class<*> = java.util.Date::class.java
    },

    UNSUPPORTEDTYPE {
        override fun typeName(): String = "UnsupportedType"

        override fun javaType(): Class<*> = java.lang.Object::class.java
    },

    URL {
        override fun typeName(): String = "URL"

        override fun javaType(): Class<*> = java.lang.String::class.java
    },

    ID {
        override fun typeName(): String = "ID"

        override fun javaType(): Class<*> = java.lang.Integer::class.java
    }
}
